src/explore-education-statistics-common/src/modules/charts/components/utils/getReferenceLineLabelPosition.ts

```diff
@@ -0,0 +1,47 @@
+import { Axis, AxisType } from '@common/modules/charts/types/chart';
+import { AxisDomain, ViewBox } from 'recharts';
+
+export default function getReferenceLineLabelPosition({
+  axis,
+  axisType,
+  otherAxisDomain = [0, 0],
+  otherAxisPosition,
+  viewBox,
+}: {
+  axis: Axis;
+  axisType: AxisType;
+  otherAxisDomain?: [AxisDomain, AxisDomain];
+  otherAxisPosition?: number;
+  viewBox?: ViewBox;
+}) {
+  const { height = 0, width = 0, x = 0, y = 0 } = viewBox ?? {};
+  const defaultXPosition = axis === 'x' ? x : width / 2 + x;
+  const defaultYPosition = axis === 'y' ? y : height / 2 + y;
+  // otherAxisPosition is set as a percentage on minor axis lines so max is 100.
+  const maxValue = axisType === 'minor' ? 100 : (otherAxisDomain[1] as number);
```

Comment:
Would do a check to ensure that `otherAxisDomain[1]` is a number before casting it. 

We'll need to handle it appropriately if it isn't. Not sure if this means we want to throw an error or change it in some way, will leave that for you to decide :wink:

Reply:
This is now handled using parseNumber in CustomReferenceLineLabel, that'll set it to undefined if not a number, then in getReferenceLineLabelPosition it'll return the default position if either domain value is undefined.